src/sentry/web/frontend/oauth_token.py

```diff
@@ -93,34 +115,31 @@ def _get_open_id_token(self, grant, request):
             return open_id_token.get_encrypted_id_token(grant=grant)
         return None
 
-    def _get_refresh_token(self, request, client_id):
-        refresh_token = request.POST.get("refresh_token")
+    def _get_refresh_token(self, request: Request, application: ApiApplication) -> HttpResponse:
+        refresh_token_code = request.POST.get("refresh_token")
         scope = request.POST.get("scope")
 
-        if not refresh_token:
+        if not refresh_token_code:
             return self.error(request, "invalid_request")
 
         # TODO(dcramer): support scope
         if scope:
             return self.error(request, "invalid_request")
 
         try:
-            application = ApiApplication.objects.get(
-                client_id=client_id, status=ApiApplicationStatus.active
+            refresh_token = ApiToken.objects.get(
+                application=application, refresh_token=refresh_token_code
             )
-        except ApiApplication.DoesNotExist:
-            return self.error(request, "invalid_client", "invalid client_id")
-
-        try:
-            token = ApiToken.objects.get(application=application, refresh_token=refresh_token)
         except ApiToken.DoesNotExist:
             return self.error(request, "invalid_grant", "invalid refresh token")
 
-        token.refresh()
+        refresh_token.refresh()
 
-        return self._process_token_details(token)
+        return self._process_token_details(token=refresh_token)
```

Comment:
hmmm.... same here.

so we have `_get_refresh_token` and `_get_access_token` 🤔 
both calling `_process_token_details`...

If we flattened the responses to return a dict into the `post` 
we could then have the `process token` step happen in the `post` as well after the error responses 🤷

Reply:
I think getting the token then processing it in either case is fine - what's the benefit of doing it like this?

Reply:
It keeps all the logic up at the top level and makes it explicitly clear / responsibility of each helper contained to a single level.
This particular case isn't too terrible since it's only 3 layers deep, but it's a slippery slope.

For this case, we're also now calling `_process_token_details` in two places, where we could just be calling it in a single location

something like
```
def post(self):
    if x:
        token_data = get_refresh_token()
    else:
        token_data = get_access_token()
    
    token_details = process_token_details(token_data)
    return Response(token_details)
```